pkg/yaml/yaml.go

```diff
@@ -32,20 +33,31 @@ func ParseObjects(manifest string) ([]*unstructured.Unstructured, error) {
 	r := strings.NewReader(manifest)
 	decoder := yaml.NewYAMLReader(bufio.NewReader(r))
 	ret := []runtime.Object{}
+	nullResult := []byte("null")
+
 	for {
-		bytes, err := decoder.Read()
+		// This reader will return a single K8s resource at the time based on the --- separator
+		objManifest, err := decoder.Read()
 		if err == io.EOF {
 			break
 		} else if err != nil {
 			return nil, err
 		}
-		if len(bytes) == 0 {
+		if len(objManifest) == 0 {
 			continue
 		}
-		jsondata, err := yaml.ToJSON(bytes)
+		jsondata, err := yaml.ToJSON(objManifest)
 		if err != nil {
 			return nil, err
 		}
+
+		// It is also possible that the provided yaml file is empty from the point of view
+		// of the toJSON parser. For example if the yaml only contain comments.
```

Comment:
Is this the case also if the file is empty (no comments)? If so, we may be able to remove https://github.com/kubeapps/kubeapps/pull/500/files#diff-dd4a0c616f20a3a6c36aab8472112d17R46 and only have the check here.

Reply:
Possibly let me check.

Reply:
Yes, it does.

I have added a test case to describe it.